### PR TITLE
[chore]: enable go-require rule from testifylint

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

Testifylint is a linter that provides best practices with the use of testify.

This PR enables [go-require](https://github.com/Antonboom/testifylint?tab=readme-ov-file#go-require) rule from [testifylint](https://github.com/Antonboom/testifylint)

### CODE REVIEW

File: exporter/datadogexporter/internal/hostmetadata/metadata_test.go [hunk 1]
```diff
@@ -190,13 +190,13 @@ func TestPushMetadata(t *testing.T) {
 		assert.Equal(t, "apikey", r.Header.Get("DD-Api-Key"))
 		assert.Equal(t, "otelcontribcol/1.0", r.Header.Get("User-Agent"))
 		reader, err := gzip.NewReader(r.Body)
-		require.NoError(t, err)
+		assert.NoError(t, err)
```
Comment (yurishkuro) [inline exporter/datadogexporter/internal/hostmetadata/metadata_test.go:193]: I'm a bit confused by the change. We already had a linter that was forcing the use of require when checking errors, sometimes incorrectly (eg you don't want require in deferred functions). But then in this case require was correct because we don't want to proceed checking return values if there was an error. 

So 
- is the linter telling that this line needs to be fixed?
- what are we changing about the linter, since we already had one

**Review by mx-psi — COMMENTED** [review]
Do you have a link to the docs explaining what this rule does?

File: .golangci.yml [hunk 2]
```diff
@@ -178,3 +178,12 @@ issues:
...
         - gosec
+    
+  # Maximum issues count per one linter.
+  # Set to 0 to disable.
+  # Default: 50
+  max-issues-per-linter: 0
+  # Maximum count of issues with the same text.
+  # Set to 0 to disable.
+  # Default: 3
+  max-same-issues: 0
```
Comment (mx-psi) [inline .golangci.yml:189]: Do we need this now? I would prefer to do it in a different PR if so

**Review by mx-psi — APPROVED** [review]
:facepalm: sorry, hadn't seen the link in the description. Just one more question, otherwise LGTM
